### PR TITLE
Don't use the previous URL if current virtual URL is safebrowsing

### PR DESCRIPTION
**Pull request by bsclifton**
Fixes https://github.com/brave/browser-laptop/issues/10143

Auditors: @diracdeltas, @alexwykoff

Test Plan:
1. open new tab
2. navigate to downloadme.org
3. notice that URL is properly "about:safebrowsing#downloadme.org"

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/browser/tabs.js [hunk 1]
```diff
@@ -302,7 +302,8 @@ const fixDisplayURL = (navigationEntry, controller) => {
     navigationEntry.virtualURL = getSourceAboutUrl(navigationEntry.virtualURL)
   }
 
-  if (isIntermediateAboutPage(navigationEntry.virtualURL)) {
+  if (isIntermediateAboutPage(navigationEntry.virtualURL) &&
+    navigationEntry.virtualURL.indexOf('about:safebrowsing#') === -1) {
```
Comment (diracdeltas) [inline app/browser/tabs.js]: if this is an issue for about:safebrowsing, shouldn't it also be an issue for about:error, about:certerror, etc?

Comment (diracdeltas) [inline app/browser/tabs.js]: maybe not because those errors fire after the page has started loading already but about:safebrowsing fires before

Comment (diracdeltas) [inline app/browser/tabs.js]: i think this should do a more strict check, like `!virtualUrl.startsWith...`

Comment (bsclifton) [inline app/browser/tabs.js]: @diracdeltas I'm not sure, maybe you can help me understand

If you open a new tab and then nav to downloadme.org, filtering picks it up via https://github.com/brave/browser-laptop/blob/0.18.x/app/filtering.js#L150 and restarts the nav process with "about:safebrowsing#downloadme.org", causing the previous page to be the full chrome-extension://about-newtab.html

If you navigate to a site and turn off your network card while the connection is happening, I would expect the browser to have an error page... but have the proper URL in the url bar (ex: the site I'm trying to reach). I was under the impression that a redirect happens after the site is unreachable (so the previous entry would be the correct site)

Comment (bsclifton) [inline app/browser/tabs.js]: doh- missed your previous comments 😛  (needed to refresh). Let me implement a better check 😄

**Comment by bsclifton** [comment]
@diracdeltas updated 😄

**Pull request merged**